cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,48 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
+import SignUpPageObject from '../support/pages/signUp.pageObject';
+
+const signUpPage = new SignUpPageObject();
+
 describe('Sign Up page', () => {
-  before(() => {
+  let user;
+
+  beforeEach(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
+
+    signUpPage.visit();
+  });
+
+  it('should provide an ability to sign up', () => {
+    signUpPage.typeUsername(user.username);
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpBtn();
+    signUpPage.checkSuccess('Welcome!');
```

Comment:
Ensure that the `checkSuccess` method in the `SignUpPageObject` correctly verifies the success message 'Welcome!' after a successful sign-up. This is crucial for the test to confirm successful registration.